completers/git_completer/cmd/root.go

```diff
@@ -57,4 +62,60 @@ func init() {
 		}
 		return action
 	})
+
+	addAliasCompletion(rootCmd)
+}
+
+func addAliasCompletion(cmd *cobra.Command) {
+	if c, _, err := cmd.Find([]string{"_carapace"}); err == nil {
+		c.PreRun = func(cmd *cobra.Command, args []string) {
+			// only inject if actually completing
+			if len(args) < 2 || args[1] != rootCmd.Name() {
+				return
+			}
+
+			// pass through args related to config
+			rootCmd.ParseFlags(args[2:])
+			gitArgs := []string{}
+			if f := rootCmd.Flag("C"); f.Changed {
+				gitArgs = append(gitArgs, "-C", f.Value.String())
+			}
+			if f := rootCmd.Flag("git-dir"); f.Changed {
+				gitArgs = append(gitArgs, "--git-dir", f.Value.String())
+			}
+
+			if aliases, err := git.Aliases(gitArgs); err == nil {
+				for key, value := range aliases {
+					// don't clobber existing commands
```

Comment:
https://git-scm.com/docs/git-config#Documentation/git-config.txt-alias states
> To avoid confusion and troubles with script usage, aliases that hide existing Git commands are ignored.